ddtrace/span.py

```diff
@@ -214,6 +217,12 @@ def to_dict(self):
         if self.span_type:
             d['type'] = self.span_type
 
+        if self._sampling_priority is not None:
```

Comment:
if that attribute stays "internal" it's fine to not use functions / properties; do we plan to never expose that to users?

Reply:
My position is: right now we use a `_sampling_priority_v1` metric instead of the `sampling.priority` tag that would ensure OT compatibility, so my guess is all this is beta, therefore fine to keep private, we can make it explicitly public later. There are subtles implications of using an attribute it, one is: it's fine to set the service/error state later than span creation as long as no distributed child span has been created, this implies control on when `_sampling_priority` is accessed. Similarly, setting `_sampling_priority_v1` after a child, distributed trace has been triggered, will lead to clear plain bugs as the information has already be sent. All in all, this is really a use-at-your-risk member, we might some day harden it, but my expectation this should come along with OT support, typically.

Reply:
totally agree! also this simplifies the implementation